File: shoring/blackjackEngine/src/main/java/com/panopset/blackjackEngine/Card.java
package com.panopset.blackjackEngine;

public class Card {

  private final CardDefinition cd;
  
  private final boolean isBlue;

  public Card(final CardDefinition cardDefinition) {
    this(cardDefinition, true);
  }

  public Card(final CardDefinition cardDefinition, boolean isBlue) {
    cd = cardDefinition;
    this.isBlue = isBlue;
  }

  private boolean isShowing = false;

  public boolean isShowing() {
    return isShowing;
  }

  public void show() {
    isShowing = true;
  }

  public Face getFace() {
    return cd.getFace();
  }

  public Suit getSuit() {
    return cd.getSuit();
  }
  
  public CardDefinition getCardDefinition() {
    return cd;
  }

  public String name() {
    return cd.name();
  }
  
  @Override
  public String toString() {
    return name();
  }

  public boolean isBlue() {
    return isBlue;
  }
}

File: code/shoring/blackjackEngine/src/main/kotlin/com/panopset/blackjackEngine/Card.kt
package com.panopset.blackjackEngine

class Card(val cardDefinition: CardDefinition, val isBlue: Boolean = true) {
    var isShowing = false

    val face: Face
        get() = cardDefinition.face
    val suit: Suit
        get() = cardDefinition.suit

    fun name(): String {
        return cardDefinition.name
    }

    override fun toString(): String {
        return name()
    }
}
